### PR TITLE
FSET-834: Progress status dates added

### PR DESCRIPTION
**Pull request by satishkamatam**

### CODE REVIEW

**Comment by henricook** [comment]
Hi @satishkamatam - there's some conflicts here

**Comment by henricook** [comment]
Are there any plans to migrate things currently in the progress-status-dates field?

**Comment by satishkamatam** [comment]
Not sure about the plans for migration but the service is updated to look up the status date from progress-status-dates or progress-status-timestamp.

File: it/repositories/ApplicationRepositorySpec.scala [hunk 1]
```diff
@@ -115,8 +116,11 @@ class ApplicationRepositorySpec extends MongoRepositorySpec {
         appStatus <- applicationRepo.findStatus(app.applicationId)
       } yield appStatus).futureValue
 
+      val hasCorrectStatusDate = Seconds.secondsBetween(DateTime.now(), applicationStatus.statusDate.get)
+        .isLessThan(Seconds.ONE)
```
Comment (henricook) [inline it/repositories/ApplicationRepositorySpec.scala]: How reliable was this? Do you think a one second buffer is enough especially where mongo is networked etc (e.g. in build?)

Comment (henricook) [inline it/repositories/ApplicationRepositorySpec.scala]: I'd perhaps consider boosting it to 5 to dodge any potential build breakage, this is my last comment on this PR

Comment (satishkamatam) [inline it/repositories/ApplicationRepositorySpec.scala]: done

Comment (henricook) [inline it/repositories/ApplicationRepositorySpec.scala]: pushed?

File: app/services/testdata/AssessmentScoresEnteredStatusGenerator.scala [hunk 2]
```diff
@@ -55,8 +55,8 @@ trait AssessmentScoresEnteredStatusGenerator extends ConstructiveGenerator {
     for {
       candidateInPreviousStatus <- previousStatusGenerator.generate(generationId, generatorConfig)
       _ <- aasRepository.save(getScoresAndFeedback(candidateInPreviousStatus.applicationId.get))
-      _ <- aRepository.updateStatus(candidateInPreviousStatus.applicationId.get, ApplicationStatuses.AssessmentScoresEntered)
-      _ <- aRepository.updateStatus(candidateInPreviousStatus.applicationId.get, ApplicationStatuses.AssessmentScoresEntered)
+      _ <- aRepository.updateStatus(candidateInPreviousStatus.applicationId.get, ASSESSMENT_SCORES_ACCEPTED)
```
Comment (satishkamatam) [inline app/services/testdata/AssessmentScoresEnteredStatusGenerator.scala]: 👍 

@henricook Have a question here any idea why we do update status twice in this testdatagenerator

Comment (henricook) [inline app/services/testdata/AssessmentScoresEnteredStatusGenerator.scala]: No idea, and it's an old test generator we'll likely delete or heavily modify for Faststream so i wouldn't worry too much!

File: it/repositories/ApplicationRepositorySpec.scala (hunk 1 above)
Comment (satishkamatam) [inline it/repositories/ApplicationRepositorySpec.scala]: all pushed

**Comment by henricook** [comment]
Glad you did this refactor SK, round of applause!

**Pull request merged**